.gitlab-ci.yml

```diff
@@ -51,10 +51,8 @@ trigger-agent-build:
   image: $VALIDATE_AGENT_BUILD
   # Exclude release branches without dependency change
   rules:
-    - if: ($CI_COMMIT_BRANCH =~ /^7.[0-9]{2}.x/)
-      changes:
-        - .deps/*
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Is it expected to not limit anymore to changes in `.deps/*`?

Reply:
The removal here is for branches that match the "release branch pattern"; this trigger is only meant for validating PR's (which will be merged either into main or into a release branch).

What I gather (from e.g. [this](https://github.com/DataDog/integrations-core/pull/9312), which implemented the ad-hoc logic that checks if the deps have actually changed) is that on Gitlab we don't have a reliable way of determining whether a file has changed. I think that's also related to all the git tricks that this job tries to use to determine which branch is "the target" (gitlab doesn't know which branch we want to merge against).